xilem/src/lib.rs

```diff
@@ -1,8 +1,100 @@
 // Copyright 2024 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
-//! An experimental Rust native UI framework.
-
+//!
+//! <div align="center" style="margin-bottom: 1em">
+//!
+//! <h1 style="margin: 0; line-height: 1em">Xilem</h1>
+//! <h6 style="margin-top: 0">An experimental Rust native UI framework</h6>
+//!
+//! [Linebender Website] · [Github] · [Community in the Xi Zulip] · [Examples]
+//!
+//! </div>
+//!
+//! Xilem is a UI toolkit. It combines ideas from `Flutter`, `SwiftUI`, and `Elm`. Like all of these, it uses lightweight view objects, diffing them to provide minimal updates to a retained UI. Like `SwiftUI`, it is strongly typed. For more details on Xilem's reactive architecture see Xilem: an [architecture for UI in Rust].
+//!
+//! Xilem's reactive layer is built on top of a wide array of foundational Rust UI projects, e.g.:
+//!
+//! * Widgets are provided by Masonry, which is a fork of the now discontinued `Druid` UI toolkit.
+//! * Rendering is provided by `Vello`, a high performance GPU compute-centric 2D renderer.
+//! * GPU compute infrastructure is provided by wgpu.
+//! * Text support is provided by `Parley`, `Fontique`, `Swash`, and `Skrifa`.
+//! * Accessibility is provided by `AccessKit`.
+//! * Window handling is provided by `winit`.
+//!
+//! Xilem can currently be considered to be in an alpha state. Lots of things need improvements.
+//!
+//! ## Example
+//! The simplest app looks like this:
+//! ```rust
+//! use winit::error::EventLoopError;
+//! use xilem::view::{button, flex, label};
+//! use xilem::{EventLoop, WidgetView, Xilem};
+//!
+//! #[derive(Default, Debug)]
+//! struct AppState {
+//!     num: i32,
+//! }
+//!
+//! fn app_logic(data: &mut AppState) -> impl WidgetView<AppState> {
+//!     flex((label(format!("{}", data.num)), button("increment", |data: &mut AppState| data.num+=1)))
+//! }
+//!
+//! fn main() -> Result<(), EventLoopError> {
+//!     let app = Xilem::new(AppState::default(), app_logic);
+//!     app.run_windowed(EventLoop::with_user_event(), "Counter".into())?;
```

Comment:
A trick we've used elsewhere is:
```suggestion
//!     # return;
//!     app.run_windowed(EventLoop::with_user_event(), "Counter".into())?;
```
which ensures that the doctest doesn't try to actually run the app.

Reply:
Just do: "rust,no_run" after the triple ticks to open the code block if you don't want it to run and just want it to compile.

Reply:
Oh yeah, that's better!